string message
int32 id_component